### ISSUE TITLE
Apothecary - FreeImage 3.17.0

### ISSUE THREAD

**Issue by danoli3** [openframeworks/openFrameworks#3925]
The Apothecary Script is now building against 3.17.0 which was released in April.
For 0.9.0, oF will be required to step up to 3.17.0 due to limitations of release builds of 64bit versions for Windows.

For other platforms: I've fixed the core issues with FreeImage that were patched for 3.16.0 for iOS / OSX.
https://github.com/danoli3/freeimage (PR's still out in the wild for a lot of the original dependancy issues), build working so far).

### FreeImage 3.17.0 current build status:

| | osx | vs | win_cb | ios | android |
|-----|-----|-------|---------|-------------|-------------|----|--------|-----|---------|
| Merged? |   ✓      | ✓  | ✓      | ✓   | ✓       |
| PR |  #3936      | #3895  |       |  #3934  |        |

**Comment by danoli3** [comment]
Hey @arturoc I just noticed this android.zip https://github.com/openframeworks/openFrameworks/blob/master/libs/FreeImage/lib/android.zip

**Comment by borg** [comment]
Hola  @arturoc,
just flagging the revert in FreeImage.h from 

#define BOOL int32_t 
to 
typedef int32_t BOOL;

causes clashes with Obj C BOOL on OSX and IOS

https://github.com/openframeworks/openFrameworks/issues/76

https://github.com/openframeworks/openFrameworks/pull/1889

Can Apothecary create separate headers for those?

**Comment by arturoc** [comment]
there's no need to have 2 headers we can patch it in the apothecary scripts with an ifdef for ios, i think something like:

```
#if TARGET_OS_IPHONE_SIMULATOR || TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE || TARGET_IPHONE
...
#endif
```

@danoli3 i've just deleted the zip, didn't saw this before

**Comment by danoli3** [comment]
@borg Yeah @bakercp just found the same issue! I think newer compilers can work this out but 10.8 and possibly 10.9 have this issue, so I'll patch the FreeImage repo directly with the fix.
